### PR TITLE
The Mulligan (traitor item)

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
A syringe which completely randomizes your name and appearance.

4tc

:cl: Kor
rscadd:  Traitors can now buy a Mulligan for 4 telecrystals, a syringe that completely randomizes your name and appearance.
rscadd:  The spy bundle now includes a switchblade and Mulligan.
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
Consider adding this to a stealth bundle as well. This is obviously meant to be used with an agent card.

**Comment by KorPhaeron** [comment]
Which file are bundles in I am lazy and dont want to look

**Comment by Bawhoppen** [comment]
As meme as it is, give it some kind of a normal name.
It could be anything. Metascopic Syringe or something.

**Comment by LanCartwright** [comment]
had an idea for the mulligan. 
Once you change your identity, you can never go back to the previous one (or alternatively, your original identity).
Trying to do so makes you unrecognisable. 
how about it. 

based it off Hannibal Bates from Flash and Arrow TV series.

**Comment by Robustin** [comment]
It makes more sense that Syndicate gear might have nicknames given that its a criminal organization filled with hardened killers whose tradecraft would have them giving weapons and tools a shorter more relatable name than the technical name. 

Think Tommy Gun v. .45 Caliber Thompson Submachine Gun

The "mulligan" sounds like an identifier used by field agents. Anything that makes it easier to communicate (and more intuitive) is better. 

Should work nicely with re-writable agent ID's, though I worry about it treading too hard on changeling territory, ultimately it will still go unused due to the high cost (why pay 20% of your TC for something genetics can do at roundstart, 6 TC gets you an emag that lets you waltz into genetics for a new identity/appearance along with the 500 other things an emag lets you do.

**Comment by Jazaen** [comment]
Only two ways to get your apperance back right now are through the GLORIOUS SUB-DEPARTMENT OF GENETICS (with previous save to cloner/machine/disc, maybe though monkeyfication and back) or using botany pods with some of your blood.

Make it so that The Mulligan can also change between human and lizard, for something genetisc can't do,

**Comment by Incoming5643** [comment]
fun idea but there's a decent chance that you'll end up as a black man with pastel blue hair and a pink beard. Which is probably not going to pass for "average crewmember" anytime soon.

**Comment by Bawhoppen** [comment]
@Incoming5643 have you seen players nowadays

**Comment by Iamgoofball** [comment]
@Incoming5643 you are very wrong here good sir

**Comment by CosmicScientist** [comment]
:+1:, I want to play a round with this badly.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -379,6 +379,20 @@
...
+	color = "#5EFF3B" //RGB: 94, 255, 59
+	metabolization_rate = INFINITY
+
+/datum/reagent/mulligan/on_mob_life(mob/living/carbon/human/H)
+	..()
+	H << "<span class='warning'><b>You grit your teeth in pain as your body rapidly mutates!</b></span>"
+	H.visible_message("<b>[H]</b> suddenly transforms!")
+	randomize_human(H)
+	return 1
+
```
Comment (KazeEspada) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: What happens with 100 units?

Comment (KorPhaeron) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: >metabolization_rate = INFINITY

Same thing that happens with 1 unit. You mutate once and its done.

**Comment by optimumtact** [comment]
mulligan virus please

**Comment by Stenthus** [comment]
sounds good, if people end up never using it maybe lower the cost.

**Comment by Cheridan** [comment]
don't see this going over very well, since there are already people who valid anyone who isn't on the crew manifest.

aside from that its ok

**Comment by KorPhaeron** [comment]
It is probably not impossible to have it generate a new security record for them, and it'd be a clearly defined class of its own vs just stealing a genetics syringe.

**Comment by bgobandit** [comment]
or you can just find a way to make a security record like a real agent

what are you, unrobust children

**Comment by tkdrg** [comment]
is this ready or not

**Comment by KorPhaeron** [comment]
@tkdrg 

Other than balance concerns sure, I can always update it later if it turns out to be shit

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 2]
```diff
@@ -11,6 +11,8 @@
 			new /obj/item/device/camera_bug(src)
 			new /obj/item/device/multitool/ai_detect(src)
 			new /obj/item/device/encryptionkey/syndicate(src)
+			new /obj/item/weapon/reagent_containers/syringe/mulligan(src)
+			new /obj/item/weapon/switchblade
```
Comment (tkdrg) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: needs a `(src)` here
also aren't 9 items a bit too much, do these even fit in a single box

Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: I think you can cram infinite items into a box if you override it cause we have infinite bagslots now, it just wont let you put them back in if you take them out.

Also you spend literally all your TC on the box it needs to give you at least one shitty weapon.

Comment (tkdrg) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: k sure, just add the (src) then whatever

**Comment by optimumtact** [comment]
update to latest master
git rebase -Xignore-all-whitespace upstream/master

**Comment by tkdrg** [comment]
don't touch this i'll just manualmerge it

**Pull request closed without merging**